relay-server/src/utils/dynamic_sampling.rs

```diff
@@ -211,6 +215,46 @@ mod tests {
         }
     }
 
+    /// ugly hack to build an envelope with an optional trace context
+    fn new_envelope(with_trace_context: bool) -> Envelope {
```

Comment:
Agreed, I toyed with the idea making some members public in order to construct what I needed easier, but I decided that it would need a bit more careful consideration and went for a brute force approach in tests.

Reply:
Yeah, rather than making members public we should create a good interface. We'll need similar facilities for testing metrics, so I can look into that.